swcc/swcc/models/project.py

```diff
@@ -66,96 +71,187 @@ def load_data_from_json(self, file, interpret):
+                        groomed_shape = self.project.add_groomed_segmentation(
+                            file=relative_path(value),
+                            segmentation=original_shape,
+                        )
+                elif key == 'local':
+                    local_particles_path = relative_path(value)
+                elif key == 'world':
+                    world_particles_path = relative_path(value)
+                elif key == 'alignment':
+                    transform = Path(temp_dir) / 'transform'
+                    with transform.open('w') as f:
+                        f.write(value)
+                elif key == 'landmarks':
+                    Landmarks(
+                        file=relative_path(value),
+                        subject=subject,
+                    ).create()
+                elif key == 'constraints':
+                    constraints_path = relative_path(value)
+                # elif key == 'procrustes':
+                #     pass
+
+            if world_particles_path or local_particles_path:
+                groomed_mesh = None
+                groomed_segmentation = None
+                if type(original_shape) == Mesh:
+                    groomed_mesh = groomed_shape
+                elif type(original_shape) == Segmentation:
+                    groomed_segmentation = groomed_shape
+                particles = OptimizedParticles(
+                    world=world_particles_path,
+                    local=local_particles_path,
+                    transform=transform,
+                    groomed_segmentation=groomed_segmentation,
+                    groomed_mesh=groomed_mesh,
+                    project=self.project,
+                ).create()
+                if constraints_path:
+                    Constraints(
+                        file=constraints_path, subject=subject, optimized_particles=particles
+                    ).create()
+
+    def download_all(self, location):
+        def relative_path(filepath):
+            return Path(
+                location,
+                str('/'.join(filepath.split('/')[:-1])).replace('../', '').replace('./', ''),
+            )
+
+        def relative_download(file, resolve):
+            if not file:
+                return
+            file.download(relative_path(resolve))
+
+        relative_download(self.project.file, '')
+        data, optimize = self.load_data(interpret=False)
+
+        for entry in data:
+            row = {}
+            for key in entry.keys():
+                prefixes = [p for p in expected_key_prefixes if key.startswith(p)]
+                if len(prefixes) > 0:
+                    row[prefixes[0]] = entry[key]
+
+            for key, value in row.items():
+                match_name = Path(value).name
+                if key == 'shape':
+                    key = shape_file_type(Path(value)).__name__.lower()
+
+                if key == 'mesh':
```

Comment:
When we have big if/elif blocks like this, it feels like we should be able to reduce the code by having a data structure and a single block of code.  If you think it would be more maintainable, you could do something like:

```
downloadTypes = {
  'mesh': ((''dataset', 'meshes'), 'file'),
  ...
  'world': (('particles', ), 'world'),
}
if key in downloadTypes:
  projattr, fileattr = downloadTypes[key]
  filelist = self.project
  for attr in projattr:
    filelist = getattr(filelist, attr)
  for f in filelist:
    if str(getattr(f, fileattr)) == match_name:
      relative_download(getattr(f, fileattr), value)
```

If you think this just makes it harder to understand, leave things as they are.